pkg/appmanager/outputConfig.go

```diff
@@ -18,17 +18,37 @@ package appmanager
 
 import (
 	"encoding/json"
+	"fmt"
 	"strings"
 	"time"
 
 	log "github.com/F5Networks/k8s-bigip-ctlr/pkg/vlogger"
 )
 
+// invalidAgent is returned when no matching agent values are given neither "as3" nor "cccl"
+var invalidAgentParam = "Invalid agent parameter provided (agent parameter can only be 'as3' or 'cccl')"
+
 // Dump out the Virtual Server configs to a file
-func (appMgr *Manager) outputConfig() {
+func (appMgr *Manager) outputConfig() error {
 	appMgr.resources.Lock()
```

Comment:
as opened a lock, add `defer appMgr.resoures.Unlock()` after immediately calling Lock(), instead of calling it before each return.

Reply:
Done changes as suggested